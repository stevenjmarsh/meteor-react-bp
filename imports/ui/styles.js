import { css } from "glamor";

/**
 *  Component Style Pattern (use with glamor)
 *
 *  const componentRules = {
 *    rule1: "value",
 *      ...
 *    ruleN: "value",
 *  };
 *  export const componentStyle = {
 *    rules: componentRules,
 *    className: `${css(componentRules)}`,
 *  };
 *
 */

const gradientBackgroundRules = {
  background: [
    "#048ec5",
    "-moz-linear-gradient(-45deg, #048ec5 0%, #2ecc71 100%) fixed",
    "-webkit-linear-gradient(-45deg, #048ec5 0%,#2ecc71 100%) fixed",
    "linear-gradient(135deg, #048ec5 0%,#2ecc71 100%) fixed",
  ],
  /* IE6-9 fallback on horizontal gradient */
  filter: "progid:DXImageTransform.Microsoft.gradient( startColorstr='#048ec5', endColorstr='#2ecc71',GradientType=1 )",
};

const defaultPageRules = {
  ...gradientBackgroundRules,
  color: "white",
  padding: "30px",
};
export const defaultPageStyle = { // eslint-disable-line import/prefer-default-export
  rules: defaultPageRules,
  className: `${css(defaultPageRules)}`,
};
