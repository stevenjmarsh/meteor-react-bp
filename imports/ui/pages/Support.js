import React from "react";
import { defaultPageStyle } from "../styles";

const Support = () => (
  <div className={defaultPageStyle.className}>
    <h1>Support</h1>
    <p>Under Construction.</p>
  </div>
);

export default Support;
