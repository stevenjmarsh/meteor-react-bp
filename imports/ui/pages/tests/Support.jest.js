/* eslint-env jest */
/* eslint-disable func-names, prefer-arrow-callback, no-unused-expressions */
/* eslint-disable import/no-extraneous-dependencies */

import React from "react";
import renderer from "react-test-renderer";
import Support from "../Support";
import {
  formattedJSON,
  getAllComponentStyle,
} from "../../../utils/tests/testUtils";
import * as styles from "../../../ui/styles";

describe("<Support />", function () {
  it("matches render snapshot", function () {
    const tree = renderer.create(<Support />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("matches style / css snapshot", function () {
    const allStyle = getAllComponentStyle(<Support />, styles);
    expect(formattedJSON(allStyle)).toMatchSnapshot();
  });
});
