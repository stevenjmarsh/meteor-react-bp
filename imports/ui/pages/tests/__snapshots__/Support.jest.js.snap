exports[`<Support /> matches render snapshot 1`] = `
<div
  className="css-wlbfil">
  <h1>
    Support
  </h1>
  <p>
    Under Construction.
  </p>
</div>
`;

exports[`<Support /> matches style / css snapshot 1`] = `
"[
  {
    \"rules\": {
      \"background\": [
        \"#048ec5\",
        \"-moz-linear-gradient(-45deg, #048ec5 0%, #2ecc71 100%) fixed\",
        \"-webkit-linear-gradient(-45deg, #048ec5 0%,#2ecc71 100%) fixed\",
        \"-webkit-linear-gradient(135deg, #048ec5 0%,#2ecc71 100%) fixed\",
        \"-moz-linear-gradient(135deg, #048ec5 0%,#2ecc71 100%) fixed\",
        \"linear-gradient(135deg, #048ec5 0%,#2ecc71 100%) fixed\"
      ],
      \"filter\": \"progid:DXImageTransform.Microsoft.gradient( startColorstr=\'#048ec5\', endColorstr=\'#2ecc71\',GradientType=1 )\",
      \"color\": \"white\",
      \"padding\": \"30px\"
    },
    \"className\": \"css-wlbfil\"
  }
]"
`;
